### PR TITLE
Fetch all interactions from deskcom

### PR DESCRIPTION
**Pull request by frj**
This PR implements the writing of all desk.com interactions to s3. This includes:
- Refactoring desk.com client and fetcher for the 'paging' scheme used by the interactions endpoint
- Exporter implementation to iterate over interaction pages
- assembly configuration for creating an executable jar of the app

### CODE REVIEW

File: application/src/main/scala/com/gu/deskcomexporttool/DeskComClient.scala [hunk 1]
```diff
@@ -29,55 +29,64 @@ object DeskComClient {
...
         }
     }
 
-    private def validateStatusCode(statusCode: Int): Either[DeskComApiError, Unit] = {
-      statusCode match {
+    private def validateStatusCode(httpResponse: HttpResponse): Either[DeskComApiError, Unit] = {
+      httpResponse.statusCode match {
         case 200 => Right(())
-        case statusCode => Left(DeskComApiError(s"Interactions endpoint returned status: $statusCode"))
+        case 422 => Left(DeskComUnprocessableEntity(httpResponse.body.take(1000)))
```
Comment (arausuy) [inline application/src/main/scala/com/gu/deskcomexporttool/DeskComClient.scala]: What is the magic 1000?

File: application/src/main/scala/com/gu/deskcomexporttool/HttpClient.scala [hunk 2]
```diff
@@ -29,7 +32,8 @@ object HttpClient {
 
       for {
         sttpResponse <- EitherT.right(sttpRequest.send())
-        body <- EitherT.fromEither(sttpResponse.body).leftMap(HttpError)
+        _ = log.debug(s"Received desk.com response status:${sttpResponse.code} body:${sttpResponse.body.right.getOrElse("").take(200)}")
```
Comment (arausuy) [inline application/src/main/scala/com/gu/deskcomexporttool/HttpClient.scala]: Magic 200?

**Review by arausuy — APPROVED** [review]

File: application/src/main/scala/com/gu/deskcomexporttool/DeskComClient.scala (hunk 1 above)
Comment (frj) [reply application/src/main/scala/com/gu/deskcomexporttool/DeskComClient.scala]: these are just truncating bodies for log messages, ill make it more explicit

File: application/src/main/scala/com/gu/deskcomexporttool/HttpClient.scala (hunk 2 above)
Comment (frj) [reply application/src/main/scala/com/gu/deskcomexporttool/HttpClient.scala]: these are just truncating bodies for log messages, ill make it more explicit

**Pull request merged**